### DISCUSSION TITLE
Objective-C backend for TeaVM

### DISCUSSION THREAD

**Discussion by Ihromant** [konsoletyper/teavm#763] [Ideas]
Currently TeaVM has C-backend (i suppose experimental and unfinished).
Was it ever considered to generate Objective-C backend (as I read in Wiki in not-objective part it's just C) with possibility to call native libraries through something similar to @JsBody annotation?
In this case TeaVM could be used to develop native applications for IOS.

**Comment by konsoletyper** [comment]
I don't think separate Objective-C backend necessary here. JSO is implemented as a plugin to TeaVM JS backend. Plugin is free to generate whatever required. There can be Objective-C plugin that generates Objective-C. Anyway, I don't have neither time, nor equipment to do that. But anyway it is possible, yes.

**Comment by konsoletyper** [comment]
BTW, C backend is not "experimental" or "unfinished". I made it to compile our application to UWP (there was no way to run Java as UWP app) and it worked, we published app to Windows store. Application is 0.5MLOC of Java/Kotlin code, so I suppose C backend can be considered quite mature, though it does not support all features of JS backend.

**Reply by Ihromant** [reply]
I suppose that in this case teavm website needs documentation section similar to "Getting started" and "Interacting with JavaScript" on teavm.org. I'm assuming that you somehow interacted with UWP platform and called some API methods.

**Reply by konsoletyper** [reply]
Sure, it does. However, have no time for that.
